packages/block-editor/src/store/selectors.js

```diff
@@ -1215,7 +1215,17 @@ export function getBlockInsertionPoint( state ) {
  * @return {?boolean} Whether the insertion point is visible or not.
  */
 export function isBlockInsertionPointVisible( state ) {
-	return state.insertionPoint !== null;
+	const insertionPoint = state.insertionPoint;
+
+	if ( ! state.insertionPoint ) {
+		return false;
+	}
+
+	if ( getTemplateLock( state, insertionPoint.rootClientId ) ) {
+		return false;
+	}
+
+	return true;
```

Comment:
I never know what's the best layer to adjust here. Should we (also or instead) prevent insertion points to be set if the template is locked? And if so, in the component or reducer? I know that in a lot of other places we're doing a lot to prevent actions from even dispatching to the store if we know it won't change anything.

The question is a bit more general, but I think it would be good to see more consistency across the code base.

Reply:
For example, for the block drop zone, we are completely disabling it when the template is locked:

https://github.com/WordPress/gutenberg/blob/05ed04a83eba9aee6e68bb3cdd99bd3a788b6311/packages/block-editor/src/components/use-block-drop-zone/index.js#L125

Reply:
So maybe we should disable the in-between inserter if the template is locked?